build.sbt

```diff
@@ -76,3 +76,10 @@ lazy val test = (projectMatrix in file("test"))
   )
   .jvmPlatform(scalaVersions = scala2)
   .jsPlatform(scalaVersions = scala2)
+
+inThisBuild(
+  List(
+    semanticdbEnabled := true,
+    semanticdbVersion := "4.4.33"
+  )
```

Comment:
Ideally we would like to enable that ad-hoc during the CI build, but it seems that right know there is no way to do that via lsif-java interface.

Reply:
does enabling semantic db have any side-effects?

also, why can't we enable this only for the duration of the build? how to w egenerate the `dump.lsif` file in the first place?

Reply:
> does enabling semantic db have any side-effects?

Yes, it might increase the build time. 

Also, I imagine that something might break some time inside of semanticDb which will cause the compilation to fail. 

> also, why can't we enable this only for the duration of the build? how to w egenerate the dump.lsif file in the first place?

We could use: https://github.com/sourcegraph/sbt-sourcegraph#other-ways-to-enable-semanticdb

but the problem is that we are not calling sbt-sourcegraph ourselves rather it is managed by lsif-java. 

One way would be to get rid of lsif-java and manually call sbt-sourcegraph but I would like to avoid that as lsif-java adds dynamically sbt-sourcegraph plugin to the project.

The other option would be to attach that information during the build using text file manipulation which would work but it's quite hacky.

Reply:
so lsif-java runs some sbt command? can't we scope the settings so that they are used only when this command is run?

Reply:
Can you scope the settings to be used for command which does not exist? Remember that java-lsif adds plugin to sbt during execution.

Reply:
hm no you can't ... I guess I'd prefer to add the plugin explicitly then. I don't want to slow compilation down (esp in sttp/tapir) or to introduce a new potential build-breaking feature, as the builds are already failing too often ;)

Reply:
Sure, that makes sens. I will update the PR.